language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/builder/impl/TypeGuardCodeAction.java

```diff
@@ -0,0 +1,208 @@
+        if (refAtCursor.getbLangNode() instanceof BLangInvocation) {
+            hasDefaultInitFunction = symbolAtCursor instanceof BObjectTypeSymbol;
+            hasCustomInitFunction = symbolAtCursor instanceof BInvokableSymbol &&
+                    symbolAtCursor.name.value.endsWith("init");
+        }
+        boolean isInitInvocation = hasDefaultInitFunction || hasCustomInitFunction;
+
+        String commandTitle;
+        try {
+            if (isInvocation || isInitInvocation) {
+                BType returnType;
+                if (hasDefaultInitFunction) {
+                    returnType = symbolAtCursor.type;
+                } else if (hasCustomInitFunction) {
+                    returnType = symbolAtCursor.owner.type;
+                } else {
+                    returnType = ((BInvokableSymbol) symbolAtCursor).retType;
+                }
+                if (returnType instanceof BUnionType) {
+                    BUnionType unionType = (BUnionType) returnType;
+                    if (!isRemoteInvocation) {
+                        // Add type guard code action
+                        commandTitle = String.format(CommandConstants.TYPE_GUARD_TITLE, symbolAtCursor.name);
+                        List<TextEdit> tEdits = getTypeGuardCodeActionEdits(context, uri, refAtCursor, unionType);
+                        if (!tEdits.isEmpty()) {
+                            return Collections.singletonList(createQuickFixCodeAction(commandTitle, tEdits, uri));
+                        }
+                    }
+                }
+            }
+        } catch (IOException | WorkspaceDocumentException e) {
+            throw new LSCodeActionProviderException("", e);
+        }
+
+        return new ArrayList<>();
+    }
+
+    private static List<TextEdit> getTypeGuardCodeActionEdits(LSContext context, String uri,
+                                                              Reference referenceAtCursor,
+                                                              BUnionType unionType)
+            throws WorkspaceDocumentException, IOException {
+        WorkspaceDocumentManager docManager = context.get(DocumentServiceKeys.DOC_MANAGER_KEY);
+        BLangNode bLangNode = referenceAtCursor.getbLangNode();
+        Position startPos = new Position(bLangNode.pos.sLine - 1, bLangNode.pos.sCol - 1);
+        Position endPosWithSemiColon = new Position(bLangNode.pos.eLine - 1, bLangNode.pos.eCol);
+        Position endPos = new Position(bLangNode.pos.eLine - 1, bLangNode.pos.eCol - 1);
+        Range newTextRange = new Range(startPos, endPosWithSemiColon);
+
+        List<TextEdit> edits = new ArrayList<>();
+        String spaces = StringUtils.repeat('\t', bLangNode.pos.sCol - 1);
+        String padding = LINE_SEPARATOR + LINE_SEPARATOR + spaces;
+        String content = CommandUtil.getContentOfRange(docManager, uri, new Range(startPos, endPos));
+        // Remove last line feed
+        while (content.endsWith("\n")) {
```

Comment:
Is it ok that we don not use the platform specific line separator?

Reply:
Yes, should use platform dependent seperator